nir/src/main/scala/scala/scalanative/nir/serialization/SerializationCaches.scala

```diff
@@ -0,0 +1,26 @@
+package scala.scalanative
+package nir
+package serialization
+
+import java.{util => ju}
+import java.lang.ref.WeakReference
+
+private[scalanative] class SerializationCaches(
+    val sharedArr128: Array[Byte],
+    val sharedArr256: Array[Byte],
+    val sharedArr512: Array[Byte],
+    val sharedArr4096: Array[Byte],
+    val internedStrings: ju.WeakHashMap[String, WeakReference[String]]
```

Comment:
A few more things to consider here:

1. Cache `Global`.
2. Cache `Sig`.

Those are often used as keys in hash maps.